### PR TITLE
Restore withdrawn [VS-581]

### PR DESCRIPTION
**Pull request by mcovarr**
Putting this out for early review as I slowly continue to test the changes. 'Hide whitespace' highly recommended when viewing the diffs due to PEP8-fueled horizontal whitespace fixes in a couple of Python scripts.

Screen Shot 2022-08-30 at 8 31 37 AM

### CODE REVIEW

File: scripts/variantstore/wdl/GvsUtils.wdl [hunk 1]
```diff
@@ -366,15 +366,17 @@ task GetNumSamplesLoaded {
...
-    set -e
+    set -o errexit -o nounset -o xtrace -o pipefail
 
     echo "project_id = ~{project_id}" > ~/.bigqueryrc
-    bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false \
-      'SELECT COUNT(*) as num_rows FROM `~{fq_sample_table}` WHERE is_loaded = true and is_control = ~{control_samples}' > num_rows.csv
+    bq query --location=US --project_id=~{project_id} --format=csv --use_legacy_sql=false '
 
-    NUMROWS=$(python3 -c "csvObj=open('num_rows.csv','r');csvContents=csvObj.read();print(csvContents.split('\n')[1]);")
+      SELECT COUNT(*) FROM `~{fq_sample_table}` WHERE
```
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsUtils.wdl:374]: I believe the input `fq_sample_table_lastmodified_timestamp` is no longer needed.

File: scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py [hunk 2]
```diff
@@ -211,109 +222,124 @@ def make_extract_table(call_set_identifier,
...
+    parser.add_argument('--fq_sample_mapping_table', type=str, help='Mapping table from sample_id to sample_name',
+                        required=True)
+    parser.add_argument('--ttl', type=int, help='Temp table TTL in hours', required=False, default=72)
+
+    sample_args = parser.add_mutually_exclusive_group(required=True)
+    sample_args.add_argument('--sample_names_to_extract', type=str,
+                             help='File containing list of samples to extract, 1 per line. ' +
+                                  'All samples in this file will be included in the cohort regardless of `withdrawn` status in the `sample_info` table.')
+    sample_args.add_argument('--fq_cohort_sample_names', type=str,
+                             help='FQN of cohort table to extract, contains "sample_name" column. ' +
```
Comment (gbggrant) [inline scripts/variantstore/wdl/extract/create_ranges_cohort_extract_data_table.py:329]: Please spell out FQN (I actually had to look it 😁 )

**Review by gbggrant — APPROVED** [review]
LGTM

File: scripts/variantstore/wdl/GvsImportGenomes.wdl [hunk 3]
```diff
@@ -348,13 +352,19 @@ task GetUningestedSampleIds {
...
-    bq --project_id=~{project_id} query --format=csv --use_legacy_sql=false ~{bq_labels} -n ~{num_samples} \
-      "SELECT sample_id, samples.sample_name FROM \`~{dataset_name}.~{table_name}\` AS samples JOIN \`${TEMP_TABLE}\` AS temp ON samples.sample_name=temp.sample_name WHERE samples.sample_id NOT IN (SELECT sample_id FROM \`~{dataset_name}.sample_load_status\` WHERE status='FINISHED')" > sample_map
+    bq --project_id=~{project_id} query --format=csv --use_legacy_sql=false ~{bq_labels} -n ~{num_samples} '
+
+      SELECT sample_id, samples.sample_name FROM `~{dataset_name}.~{table_name}` AS samples JOIN `~{temp_table}` AS temp ON
+        samples.sample_name = temp.sample_name WHERE
+          samples.sample_id NOT IN (SELECT sample_id FROM `~{dataset_name}.sample_load_status` WHERE status="FINISHED") AND
+          samples.withdrawn is NULL
+
+    ' > sample_map
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsImportGenomes.wdl:362]: csv the filename

File: scripts/variantstore/wdl/GvsImportGenomes.wdl [hunk 4]
```diff
@@ -306,33 +306,37 @@ task GetUningestedSampleIds {
...
+    # Get the current min/max id, or 0 if there are none. Withdrawn samples still have IDs so don't filter them out.
+    bq --project_id=~{project_id} query --format=csv --use_legacy_sql=false ~{bq_labels} '
 
-    # get the current maximum id, or 0 if there are none
-    bq --project_id=~{project_id} query --format=csv --use_legacy_sql=false ~{bq_labels} \
-      "SELECT IFNULL(MIN(sample_id),0) as min, IFNULL(MAX(sample_id),0) as max FROM \`~{dataset_name}.~{table_name}\` AS samples JOIN \`${TEMP_TABLE}\` AS temp ON samples.sample_name=temp.sample_name" > results
+      SELECT IFNULL(MIN(sample_id),0) as min, IFNULL(MAX(sample_id),0) as max FROM `~{dataset_name}.~{table_name}`
+        AS samples JOIN `~{temp_table}` AS temp ON samples.sample_name = temp.sample_name
+
+    ' > results
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsImportGenomes.wdl:339]: csv the filename

**Review by rsasch — APPROVED** [review]

**Pull request merged**